enterprise/frontend/src/embedding-sdk/components/public/MetabotQuestion/MetabotQuestion.stories.tsx

```diff
@@ -35,12 +33,296 @@ const Template: StoryFn<MetabotQuestionProps> = () => {
+              },
+            ],
+            history: [
+              {
+                role: "user",
+                content: "This is what user has typed",
+              },
+              {
+                content:
+                  "lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum",
+                role: "assistant",
+              },
+            ],
+            state: {
+              queries: {},
+            },
+            conversation_id: "c7635bd0-84f3-f06d-1db8-571db7ff0cf6",
+          });
+        }),
+      ],
+    },
+  },
+};
+
+export const MultipleMessagesReaction = {
+  render: Template,
+  parameters: {
+    msw: {
+      handlers: [
+        http.post("*/api/ee/metabot-v3/v2/agent", () => {
+          return HttpResponse.json({
+            reactions: [
+              {
+                type: "metabot.reaction/message",
+                "repl/message_color": "green",
+                "repl/message_emoji": "🤖",
+                message: "This is the first message",
+              },
+              {
+                type: "metabot.reaction/message",
+                "repl/message_color": "green",
+                "repl/message_emoji": "🤖",
+                message: "This is the second message",
+              },
+            ],
+            history: [
+              {
+                role: "user",
+                content: "This is what user has typed",
+              },
+              {
+                content: "This is the first message",
+                role: "assistant",
+              },
+              {
+                content: "This is the second message",
+                role: "assistant",
+              },
+            ],
+            state: {
+              queries: {},
+            },
+            conversation_id: "c7635bd0-84f3-f06d-1db8-571db7ff0cf6",
+          });
+        }),
+      ],
+    },
+  },
+};
 
-  args: {
-    questionId: QUESTION_ID,
-    isSaveEnabled: true,
-    targetCollection: undefined,
-    title: true,
-    withResetButton: true,
+export const MetabotError = {
+  render: Template,
+  parameters: {
+    msw: {
+      handlers: [
+        http.post("*/api/ee/metabot-v3/v2/agent", () => {
+          return new HttpResponse(null, {
+            status: 500,
+          });
+        }),
```

Comment:
Nice! Just curious, is there a specific reason we opted for MSW instead of the usual e2e `cy.intercept("POST")` for mocking HTTP errors?

Reply:
This is a Storybook story, not cypress 😅 . It's also great for making storybook stories more interactive without having to wait for the AI response which is usually takes a while, a few seconds or more.

Reply:
well yeah, the other way is to extract stuff to be pure UI, and test different UI there. I think this solution compromises not having a full blown testing by kind of having storybook as e2e psuedo-tests while also developing it.